src/manifolds/Multinomial.jl

```diff
@@ -0,0 +1,93 @@
+````
+where $\mathbb{1}_k$ is the vector of length $k$ containing ones.
+
+This yields exactly the same metric as
+considering the product metric of the probablity vectors, i.e. [`PowerManifold`](@ref) of the
+$(n-1)$-dimensional [`ProbabilitySimplex`](@ref).
+
+The [`ProbabilitySimplex`](@ref) is stored internally within `M.manifold`, such that all functions of
+[`AbstractPowerManifold`](@ref) can be used directly.
+
+# Constructor
+
+    MultinomialMatrices(n,m)
+
+Generate the manifold of matrices $\mathbb R^{n×m}$ such that the $m$ columns are
+discrete probability distributions, i.e. sum up to one.
+"""
+struct MultinomialMatrices{N,M,S} <: AbstractPowerManifold{ℝ,ProbabilitySimplex{S},ArrayPowerRepresentation} where {N,M}
+    manifold::ProbabilitySimplex{S}
+end
+
+MultinomialMatrices(n::Int, m::Int) =  MultinomialMatrices{n,m,n-1}(ProbabilitySimplex(n - 1))
+
+^(M::ProbabilitySimplex{N}, m::Int) where {N} = MultinomialMatrices{manifold_dimension(M)+1,m,N}(M)
+
+@doc raw"""
+    check_manifold_point(M::MultinomialMatrices,p)
+
+Checks whether `p` is a valid point on the [`MultinomialMatrices`](@ref)`{m,n}` `M`, i.e. is a matrix
+of `m` discrete probability distributions as columns from $\mathbb R^{n}$, i.e. each column is a point from
+[`ProbabilitySimplex`](@ref)`(n-1)`.
+"""
+check_manifold_point(::MultinomialMatrices, ::Any)
+function check_manifold_point(M::MultinomialMatrices{n,m}, p; kwargs...) where {n,m}
+    if size(p) != (n, m)
+        return DomainError(
+            length(p),
+            "The matrix in `p` ($(size(p))) does not match the dimensions of $(M).",
+        )
+    end
+    return check_manifold_point(PowerManifold(M.manifold, m), p; kwargs...)
+end
+@doc raw"""
+    check_tangent_vector(M::MultinomialMatrices p, X; check_base_point = true, kwargs...)
+
+Checks whether `X` is a valid tangent vector to `p` on the [`MultinomialMatrices`](@ref) `M`.
+This means, that `p` is valid, that `X` is of correct dimension and columnswise
+a tangent vector to the columns of `p` on the [`ProbabilitySimplex`](@ref).
+The optional parameter `check_base_point` indicates, whether to call [`check_manifold_point`](@ref)  for `p`.
+"""
+function check_tangent_vector(
+    M::MultinomialMatrices{n,m},
+    p,
+    X;
+    check_base_point = true,
+    kwargs...,
+) where {n,m}
+    if check_base_point && size(p) != (n, m)
+        return DomainError(
+            length(p),
+            "The matrix `p` ($(size(p))) does not match the dimension of $(M).",
+        )
+    end
+    if size(X) != (n, m)
+        return DomainError(
+            length(X),
+            "The matrix `X` ($(size(X))) does not match the dimension of $(M).",
+        )
+    end
+    return check_tangent_vector(
+        PowerManifold(M.manifold, m),
+        p,
+        X;
+        check_base_point = check_base_point,
+        kwargs...,
+    )
+end
+
+get_iterator(M::MultinomialMatrices{n,m}) where {n,m} = 1:m
```

Comment:
Would it make more sense to use `Base.OneTo(m)`? Maybe the compiler can do some optimizations if it knows the first element is 1.

Reply:
I don't see an argument against this, so yes, I'll do that.

Reply:
I doubt it would make any noticeable difference in this case. The loops where it's used are relatively expensive anyway.

Reply:
But it also does no harm, right?

Reply:
Currently my number 1 idea for speeding up `PowerManifold` is marking more methods as `@inline` to inline them into these loops in `PowerManifold` so that views that I make there could be stack-allocated.

Reply:
Sounds nice. For now I changed this line to @sethaxen|s suggestion, since I think it's at least not worse that `1:m`, and if we finde even better ways, we can change that further.

Reply:
Yes, of course, there is definitely no harm in this.